core/src/rpc_subscriptions.rs

```diff
@@ -313,25 +363,142 @@ impl RpcSubscriptions {
     }
 
     pub fn notify_slot(&self, slot: Slot, parent: Slot, root: Slot) {
-        let subscriptions = self.slot_subscriptions.read().unwrap();
-        for (_, sink) in subscriptions.iter() {
-            sink.notify(Ok(SlotInfo { slot, parent, root }))
-                .wait()
-                .unwrap();
+        self.enqueue_notification(NotificationEntry::Slot(SlotInfo { slot, parent, root }));
+    }
+
+    fn enqueue_notification(&self, notification_entry: NotificationEntry) {
```

Comment:
from @carllin:

![image](https://user-images.githubusercontent.com/153621/72641404-229eb680-3938-11ea-8493-0918bcaf5910.png)

Reply:
applied mvines suggestion to use `channel()` instead